p2p/adapters/inproc.go

```diff
@@ -243,13 +256,13 @@ func (self *SimNode) RunProtocol(node *SimNode, rw, rrw p2p.MsgReadWriter, peer
 	log.Trace(fmt.Sprintf("protocol starting on peer %v (connection with %v)", self.Id, id))
 	p := p2p.NewPeer(id.NodeID, id.Label(), []p2p.Cap{})
 	go func() {
-		self.network.DidConnect(self.Id, id)
+		self.peerFeed.Send(&p2p.PeerEvent{Type: p2p.PeerEventTypeAdd, Peer: id.NodeID})
```

Comment:
not sure why we do it on the server level. 
we need the protocol level anyway. 
Why not just do a global wrapper around the protocol.run function

```golang
func (self *p2p.Protocol) RunWithEvents() *p2p.Protocol {
    run := self.run
    self.run = func(p *p2p.Peer, rw p2p.MsgReadWriter) error { 
        srv.peerFeed.Send(&PeerEvent{Type: PeerEventTypeAdd, Peer: p.ID()})
	defer srv.peerFeed.Send(&PeerEvent{Type: PeerEventTypeDrop, Peer: p.ID()})
       return run(p, rw)
  }
}
```